ext/magic/extconf.rb

```diff
@@ -1,42 +1,152 @@
 # frozen_string_literal: true
 
+require 'find'
 require 'mkmf'
-require 'digest'
-require 'open-uri'
-
-LIBMAGIC_TAG = '5.39'.freeze
-
-workdir = Dir.pwd
-libdir = File.join(workdir, 'file-' + LIBMAGIC_TAG)
-gemdir = File.expand_path(File.join(__dir__, '../..'))
-gem_ext_dir = File.join(gemdir, 'lib', 'ext')
-gem_include_dir = File.join(gem_ext_dir, 'include')
-gem_lib_dir = File.join(gem_ext_dir, 'lib')
-build_lib_dir = File.join(libdir, 'src', '.libs')
-
-expected_sha256 = 'f05d286a76d9556243d0cb05814929c2ecf3a5ba07963f8f70bfaaa70517fad1'
-filename = "#{workdir}/file.tar.gz"
-
-unless File.exist?(filename)
-  File.open(filename, 'wb') do |target_file|
-    URI.open("https://fossies.org/linux/misc/file-#{LIBMAGIC_TAG}.tar.gz", "User-Agent" => "RubyMagic/#{RUBY_DESCRIPTION}") do |read_file|
-      target_file.write(read_file.read)
+require 'pathname'
+
+LIBMAGIC_TAG = '5.39'
+LIBIMAGE_SHA256 = 'f05d286a76d9556243d0cb05814929c2ecf3a5ba07963f8f70bfaaa70517fad1'
+
+# helpful constants
+PACKAGE_ROOT_DIR = File.expand_path(File.join(File.dirname(__FILE__), '..', '..'))
+
+# The gem version constraint in the Rakefile is not respected at install time.
+# Keep this version in sync with the one in the Rakefile !
+REQUIRED_MINI_PORTILE_VERSION = "~> 2.5.0"
```

Comment:
Might need to add `mini_portile2` to `Gemfile` too so that it would get installed on `bundle install`. Probably only useful for development, etc.

Reply:
As Nokogiri does, the `.gemspec` has a runtime dependency to `mini_portile2`, so it gets installed all the time.

Reply:
Hi @stanhu,

I saw you added it where needed, indeed. I removed everything that was in local "gemsets", and `bundle install` installed everything nicely. I am not sure what the problem was.

Sorry for the commotion!

Krzysztof